src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -244,5 +244,10 @@ public DataProviderMetadata GetMetadata()
         {
             return null;
         }
+
+        public Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(string siteName)
```

Comment:
Instead of taking this approach, I suggest we just make the changes to detectors to use the `GetRuntimeSiteSlotMap(string stampName, string siteName)` variant

Reply:
I don't see any detectors using this - it would probably be safe to remove it from the API